src/unittests/f32_interval.spec.ts

```diff
@@ -3958,14 +3959,512 @@ g.test('subtractionMatrixInterval')
     );
   });
 
-interface MatrixScalarCase {
+g.test('multiplicationMatrixMatrixInterval')
+  .paramsSubcasesOnly<MatrixPairToMatrixCase>([
+    // Only testing that different shapes of matrices are handled correctly
+    // here, to reduce test duplication.
+    // multiplicationMatrixMatrixInterval uses and transposeInterval &
+    // dotInterval for calculating intervals, so the testing for those functions
+    // will cover the actual interval calculations.
+    {
+      input: [
+        [
+          [1, 2],
+          [3, 4],
+        ],
+        [
+          [10, 20],
+          [30, 40],
+        ],
+      ],
+      expected: [
+        [70, 100],
```

Comment:
So, without spending too many cycles thinking about column / row major, and just looking at your outputs:

```
 ╭      ╮   ╭        ╮   ╭          ╮
 │ 1, 2 │ x │ 10, 20 │ = │ 70,  100 │
 │ 3, 4 │   │ 30, 40 │   │ 150, 220 │
 ╰      ╯   ╰        ╯   ╰          ╯

 dot([1, 2], [10, 30]) =  70
 dot([1, 2], [20, 40]) = 100
 dot([3, 4], [10, 30]) = 150
 dot([3, 4], [20, 40]) = 220

```

Or in symbols:

```
 ╭      ╮   ╭      ╮   ╭      ╮
 │ a, b │ x │ e, f │ = │ i, j │
 │ c, d │   │ g, h │   │ k, l │
 ╰      ╯   ╰      ╯   ╰      ╯

 dot([a, b], [e, g]) = i
 dot([a, b], [f, h]) = j
 dot([c, d], [e, g]) = k
 dot([c, d], [f, h]) = l
```

So <code>result<sub>[y,x]</sub> = dot(LHS<sub>row y</sub>, RHS<sub>column x</sub>)</code>

Reply:
I think is the annoying column major thing that I keep on hooking myself on.

[[1, 2], [3, 4]] is actually the matrix
```
 ╭      ╮   
 │ 1, 3 │
 │ 2, 4 │
 ╰      ╯
 ```
 b/c the first array `[1, 2]` is the values for column 1, so the first entries in each row.
 
For matrix product in column-major form the restriction is matKxR * matCxK, the 'outer' dimensions have to match, not the 'inner' ones, which is opposite of what I am used to, since I learnt linear algebra in row-major form
 
 When working on this, I actually turn my head sideways for it to read 'correctly' for me

Reply:
Argh, I see that the transpose of these numbers is probably what your code is doing (which given the data is column major, is correct), however the numbers you are using means either work.

Reply:
Please can you use different numbers so that `dot([a, b], [e, g])` is not equal to `dot([a, c], [e, f])`?

This will save future Ben from going down the same path.

Reply:
Updated the tests to avoid these cases.